app.py

```diff
@@ -30,24 +34,89 @@ def __init__(self):
         super().__init__()
 
     def _appmetadata(self):
-        # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._load_appmetadata
-        # Also check out ``metadata.py`` in this directory.
-        # When using the ``metadata.py`` leave this do-nothing "pass" method here.
-        pass
+        metadata = AppMetadata(
+            name="DatimexExtraction",
+            description="Extracts date and time expressions from raw text input.",
+            app_version="1.0.0",
+            app_license="Apache 2.0"
+        )
+        metadata.add_input(DocumentTypes.TextDocument)
+        metadata.add_output(AnnotationTypes.TimeExpression)
+        metadata.add_parameter(
+            name="regex",
+            description="Custom regex pattern for extracting date/time.",
+            type="string",
+            default=""
+        )
+        return metadata
 
     def _annotate(self, mmif: Mmif, **parameters) -> Mmif:
         # see https://sdk.clams.ai/autodoc/clams.app.html#clams.app.ClamsApp._annotate
-        raise NotImplementedError
+
+        if isinstance(mmif, str):
+            mmif = Mmif(mmif)
+        
+        regex_pattern = parameters.get("regex", "")
```

Comment:
I like to keep these lines here, for the day we decide to expose the custom regex as a parameter. But maybe leaving a comment for future reference be a good idea.